enterprise_subsidy/apps/content_metadata/api.py

```diff
@@ -124,21 +124,52 @@ def get_geag_variant_id_for_content(self, content_identifier, content_data):
                 variant_id = additional_metadata.get('variant_id')
         return variant_id
 
+    def enroll_by_date_for_content(self, content_data, content_mode):
+        """
+        Determines the enrollment deadline for the given ``content_data``,
+        which could be either a course or course run, hence the branching
+        on content type below.
+
+        Args:
+            content_data: A dictionary of metadata pertaining to a course or course run.
+            content_mode: The (already computed) course mode for the content.
+
+        Returns:
+            A datetime string, or possibly null.
+        """
+        if content_data.get('content_type') == 'course':
+            return course_metadata.get('normalized_metadata', {}).get('enroll_by_date')
+
+        if content_mode == CourseModes.EDX_VERIFIED.value:
+            seats = content_data.get('seats', [])
+            seats_for_mode = [
+                seat for seat in content_data.get('seats', [])
+                if seat.get('type') == content_mode
+            ]
+            if seats_for_mode:
+                seat = seats_for_mode[0]
+                return seat.get('upgrade_deadline')
```

Comment:
Should this be taking into account the `upgrade_deadline_override` field, if it has a value?

```python
return seat.get('upgrade_deadline_override') or seat.get('upgrade_deadline')
```

For example, the `federated-content-connector` cron job parses `enroll_by` date in the `CourseDetails` model (for the enrollments API supporting the Learner Portal dashboard), taking into account both `upgrade_deadline_override` and `upgrade_deadline` [here](https://github.com/edx/federated-content-connector/blob/dd483c2b3952656060033aab3daa08052b143ac8/federated_content_connector/course_metadata_importer.py#L224-L241).

I assume this method's logic should largely be parity with the logic used by `federated-content-connector` and vice versa?